pkg/processor/metricfrequencyprocessor/sieve.go

```diff
@@ -0,0 +1,197 @@
+package metricfrequencyprocessor
+
+import (
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"math"
+	"sort"
+	"time"
+)
+
+const (
+	float64EqualityThreshold = 1e-9
+	minPointAccumulationTime = time.Minute * 15
+	category1ReportFrequency = time.Minute * 5
+	category2ReportFrequency = time.Minute * 2
+	category3ReportFrequency = time.Second * 30
+	safetyInterval           = time.Second * 1
+
+	iqrAnomalyCoef            = 1.5
+	variationIqrThresholdCoef = 4
+)
+
+// MetricSieve removes data points from MetricSlices that would be reported more often than preset
+// frequency for a given category.
+// For metric sieve, there are three categories of metrics:
+// 1) Constant metrics
+// 2) Low info metrics - i.e. no anomaly in terms of iqr and low variation
+// 3) All other metrics
+type MetricSieve struct {
+	metricCache  *MetricCache
+	lastReported map[string]pdata.Timestamp
+}
+
+func (fs *MetricSieve) Sift(metric pdata.Metric) bool {
+	if metric.DataType() == pdata.MetricDataTypeDoubleGauge {
+		return fs.siftDropGauge(metric)
+	} else {
+		return false
+	}
```

Comment:
switch/case for this would suit better I believe

Reply:
yup, changed